examples/go/cmd/demo-cli/main.go

```diff
@@ -155,36 +160,39 @@ func handleEvents(session diatheke.Session) {
 		return
 	}
 
-	for {
-		// Wait for the next event from the server
-		event, err := eventStream.Recv()
-
-		// This indicates the stream has finished, which will happen
-		// when the session ends or the context used to create the stream
-		// closes.
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			textui.PrintError("Event stream error: %v\n", err)
-			return
-		}
-
-		// Check the event type
-		switch e := event.Result.(type) {
-		case *diathekepb.DiathekeEvent_Recognize:
-			handleRecognizeEvent(e.Recognize)
-
-		case *diathekepb.DiathekeEvent_Reply:
-			handleReplyEvent(e.Reply)
-
-		case *diathekepb.DiathekeEvent_Command:
-			handleCommandEvent(e.Command, session)
-
-		default:
-			textui.PrintError("Error: received unknown event type from Diatheke\n")
+	// Start a separate go routine to handle events coming from the server
+	go func() {
+		for {
+			// Wait for the next event from the server
+			event, err := eventStream.Recv()
+
+			// This indicates the stream has finished, which will happen
+			// when the session ends or the context used to create the stream
+			// closes.
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				textui.PrintError("Event stream error: %v\n", err)
+				return
```

Comment:
Is there a reason to break and return differently?  Should we make these the same?

Reply:
There was a nuanced difference before, but now that it is wrapped in the go routine, it doesn't matter. I can make them the same.